src/main/kotlin/no/nav/tag/innsynAareg/controller/RestResponseEntityExceptionHandler.kt

```diff
@@ -36,6 +42,37 @@ class RestResponseEntityExceptionHandler : ResponseEntityExceptionHandler() {
         return getResponseEntity(e, "ingen tilgang", HttpStatus.UNAUTHORIZED)
     }
 
+    @ExceptionHandler(AltinnrettigheterProxyKlientFallbackException::class)
+    @ResponseBody
+    protected fun handleAltinnFallbackFeil(
+        e: AltinnrettigheterProxyKlientFallbackException,
+        ignored: WebRequest?,
+    ): ResponseEntity<Any> {
+        if (e.cause is SocketTimeoutException) {
+            return getResponseEntity(e, "Fallback til Altinn feilet pga timeout", HttpStatus.GATEWAY_TIMEOUT)
+        }
+        val httpStatus = hentDriftsforstyrrelse(e)
+        return if (httpStatus != null) {
+            getResponseEntity(e, "Fallback til Altinn feilet pga driftsforstyrrelse", httpStatus)
+        } else {
+            handleInternalError(e, ignored)
+        }
+    }
+
+    private fun hentDriftsforstyrrelse(e: AltinnrettigheterProxyKlientFallbackException): HttpStatus? {
+        return when (e.cause) {
```

Comment:
Tror du kan skrive `val c = e.cause` i stede for `e.cause`, så får du `c` som en variabel av riktig type i det første tilfellet. (Som du gjør under med status)

Reply:
genialt :) det ble mye bedre